oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtEncoderParameters.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2002-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.jwt;
+
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+/**
+ * A holder of parameters containing the JWS headers and JWT Claims Set.
+ *
+ * @author Joe Grandja
+ * @since 5.6
+ * @see JwsHeader
+ * @see JwtClaimsSet
+ * @see JwtEncoder
+ */
+public final class JwtEncoderParameters {
+
+	private final JwsHeader jwsHeader;
+
+	private final JwtClaimsSet claims;
+
+	private JwtEncoderParameters(JwsHeader jwsHeader, JwtClaimsSet claims) {
+		Assert.notNull(claims, "claims cannot be null");
+		this.jwsHeader = jwsHeader;
+		this.claims = claims;
+	}
+
+	/**
+	 * Returns a new {@link JwtEncoderParameters}, initialized with the provided
+	 * {@link JwtClaimsSet}.
+	 * @param claims the {@link JwtClaimsSet}
+	 * @return the {@link JwtEncoderParameters}
+	 */
+	public static JwtEncoderParameters with(JwtClaimsSet claims) {
```

Comment:
I'm thinking that `with` is usually for builders and `from` is for static factory methods. Am I mistaken? If so, please disregard.

Reply:
Changed to `from()`